### ISSUE TITLE
Command line option to set temp folder

### ISSUE THREAD

**Issue by darkman97i** [jbarlow83/OCRmyPDF#440]
**Describe the issue**
When the application crashes the com.github.ocrmypdf.XXXXX temporary folder in the /tmp files is not deleted.

**To Reproduce**
Issue happens sometimes when processing huge files ( 600 plus pages and 300 plus MB )

```bash
ocrmypdf -l eng+hye --output-type pdfa Test.pdf out.pdf
```

**Example file**
Cannot include sample files, because have private information

**Expected behavior**
In my case will be enough to set the home path of the temp folder used by the application from the command line. When the process has finished, I will remove the folder ensuring there's no garbage in the OS.

**System:**
- OS: Linux mint
- OCRmyPDF Version: [e.g. v6.1.2]

**Additional context**

**Comment by jbarlow83** [comment]
You can use `env TMPDIR=/elsewhere ocrmypdf...` to change the root temporary folder. It will append the `com.github.ocrmypdf.`

As for whether to delete temporary folders after a recoverable crash (e.g. an exception), I'd prefer to not cause further problems for a system but the temporary folders can be useful in figuring out what went wrong. I can't do anything if the python process itself dies. 

Incidentally, what's the crash?

**Comment by darkman97i** [comment]
I'm with you about it is better to solve the crash. My proposal was in the way not to change the global tmp folder in the OS environment I was thinking on being able to set the specific temp folder for each ocrmypdf execution.

I will explain my scenario with more detail, really I'm not executing the application directly from the terminal, I'm calling the application from my JAVA app. When the command line executed from JAVA crash does not provide a detailed description of what happened.

I have one of this tmp folder. It contains about 14GB of files into ( yes it is a huge amount of data generated from the 300mb PDF file with 600+ pages ). I share you the current list of these folders. Tell me if there's some relevant file to watch into or some log.
[list.txt](https://github.com/jbarlow83/OCRmyPDF/files/3755444/list.txt)

**Comment by darkman97i** [comment]
More information about the issue. I have executed from the command line:
```
ocrmypdf -l eng+hye --output-type pdfa e08ecdb0-fa51-412d-a1f0-a5aca41e0dee.pdf out.pdf
WARNING -    7: [tesseract] lots of diacritics - possibly poor OCR
WARNING -   15: [tesseract] lots of diacritics - possibly poor OCR
  ERROR -   52: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -   52: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -   56: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -   56: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -   64: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -   64: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -  111: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  111: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -  111: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  111: [tesseract] Error in pixScanForForeground: invalid box
WARNING -  117: [tesseract] lots of diacritics - possibly poor OCR
   INFO -  152: [tesseract] Image too small to scale!! (3x36 vs min width of 3)
   INFO -  152: [tesseract] Line cannot be recognized!!
   INFO -  152: [tesseract] Image too small to scale!! (3x36 vs min width of 3)
   INFO -  152: [tesseract] Line cannot be recognized!!
   INFO -  152: [tesseract] Image too small to scale!! (3x36 vs min width of 3)
   INFO -  152: [tesseract] Line cannot be recognized!!
   INFO -  152: [tesseract] Image too small to scale!! (3x36 vs min width of 3)
   INFO -  152: [tesseract] Line cannot be recognized!!
WARNING -  174: [tesseract] lots of diacritics - possibly poor OCR
  ERROR -  239: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  239: [tesseract] Error in pixScanForForeground: invalid box
WARNING -  257: [tesseract] lots of diacritics - possibly poor OCR
WARNING -  261: [tesseract] lots of diacritics - possibly poor OCR
  ERROR -  297: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  297: [tesseract] Error in pixScanForForeground: invalid box
WARNING -  298: [tesseract] lots of diacritics - possibly poor OCR
WARNING -  302: [tesseract] lots of diacritics - possibly poor OCR
WARNING -  411: [tesseract] lots of diacritics - possibly poor OCR
WARNING -  468: [tesseract] lots of diacritics - possibly poor OCR
WARNING -  479: [tesseract] lots of diacritics - possibly poor OCR
  ERROR -  487: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  487: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -  523: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  523: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -  566: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  566: [tesseract] Error in pixScanForForeground: invalid box
  ERROR -  572: [tesseract] Error in boxClipToRectangle: box outside rectangle
  ERROR -  572: [tesseract] Error in pixScanForForeground: invalid box
   INFO - Output file is a PDF/A-2B (as expected)
```

But the process really does not finish in the terminal, although the pdf file has been created in the terminal the ocrmypdf process has not finished.

**Comment by jbarlow83** [comment]
Since this concerns ocrmypdf v6.1.2 (quite old now) - I imagine you're on Python 3.5, which had some bugs in multiprocessing that were fixed in 3.6. I don't think it's relevant any more. If you have information that concerns a new version please open a new issue.
